File: api/models/craft.cs
namespace Api.Model;

public class Craft
{
  public Guid CraftId { get;set; }
  public string ProductName { get;set; } = string.Empty;
  public string? Address { get;set; }
  public string? Prefecture { get;set; }
  public double? Latitude { get;set; }
  public double? Longitude { get;set; }
  public string? Image { get;set; }
  public string? Description { get;set; }
  public DateTime CreateAt { get;set; }
  public ICollection<User> Successors { get;set; } = new List<User>();
}

File: web/src/lib/type/craft.ts
import type { UserPublic } from "./user";

// GET /api/crafts
export type CraftSummary = {
  craftId: string;
  productName: string;
  address: string | null;
  prefecture: string | null;
  latitude: number | null;
  longitude: number | null;
  image: string | null;
};

// GET /api/craft/{id}
export type CraftDetail = {
  craftId: string;
  productName: string;
  address: string | null;
  prefecture: string | null;
  latitude: number | null;
  longitude: number | null;
  image: string | null;
  description: string | null;
  successorCount: number;
  successors: UserPublic[];
  createAt: string;
};
